From 455ad05d2ec3411428268c1d5dc352587016cddf Mon Sep 17 00:00:00 2001
From: Tomas Winkler <tomas.winkler@intel.com>
Date: Wed, 8 Jan 2014 20:57:44 +0200
Subject: [PATCH 290/441] mei: nfc: mei_nfc_free has to be called under lock

nfc_nfc_free unlink clients from the device list
and has to be called under mei mutex

Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
Reviewed-by: Alexander Usyskin <alexander.usyskin@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 0631213f3f6ee776db2fb2136778b570d6a17cab)

Signed-off-by: Nagaraj S <sivasankaranx.nagaraj@intel.com>
---
 drivers/misc/mei/nfc.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/misc/mei/nfc.c
+++ b/drivers/misc/mei/nfc.c
@@ -468,7 +468,9 @@ static void mei_nfc_init(struct work_str
 	return;
 
 err:
+	mutex_lock(&dev->device_lock);
 	mei_nfc_free(ndev);
+	mutex_unlock(&dev->device_lock);
 
 	return;
 }
